--- wowup-electron/src/app/addon-providers/curse-addon-provider.ts
+++ wowup-electron/src/app/addon-providers/curse-addon-provider.ts
@@ -72,13 +72,13 @@
   public readonly forceIgnore = false;
   public readonly allowChannelChange = true;
   public readonly allowReinstall = true;
   public readonly canBatchFetch = true;
   public readonly allowEdit = true;
 
-  public adRequired = true;
+  public adRequired = false;
   public enabled = true;
 
   public constructor(
     private _cachingService: CachingService,
     private _networkService: NetworkService,
     private _tocService: TocService
