--- wowup-electron/app/main.ts
+++ wowup-electron/app/main.ts
@@ -550,8 +550,8 @@
   }
   return argv.hidden || loginItems.wasOpenedAsHidden;
 }
 
 function getUserAgent() {
   const portableStr = isPortable ? " portable;" : "";
-  return `WowUp-Client/${app.getVersion()} (${osType()}; ${osRelease()}; ${osArch()}; ${portableStr} +https://wowup.io)`;
+  return `WowUp-Client/${app.getVersion()} (${osType()}; ${osRelease()}; ${osArch()}; CF; ${portableStr} +https://wowup.io)`;
 }
