--- wowup-electron/app/ipc-events.ts
+++ wowup-electron/app/ipc-events.ts
@@ -223,14 +223,8 @@
     return true;
   });
 
-  handle(IPC_OW_IS_CMP_REQUIRED, async (): Promise<boolean> => {
-    // NOTE(twolf): Next version of the ow-electron will fix the types
-    try {
-      return await (app as any).overwolf.isCMPRequired();
-    } catch (e) {
-      console.error("IPC_OW_IS_CMP_REQUIRED failed", e);
-      return false;
-    }
+  handle(IPC_OW_IS_CMP_REQUIRED, (): boolean => {
+    return true;
   });
 
   handle(IPC_OW_OPEN_CMP, (evt, cmpTab?: string) => {
@@ -238,8 +232,6 @@
     if (cmpTab) {
       options.tab = cmpTab;
     }
-
-    (app as any).overwolf.openCMPWindow(options);
   });
 
   handle(IPC_GET_ZOOM_FACTOR, () => {
